Document all public APIs in Jupyter.csproj.

cgranade opened the pull request:
This PR adds API documentation for all public APIs in the Jupyter project for IQ#.

tcNickolas commented on src/Jupyter/ConfigurationSource.cs:
```diff
@@ -13,35 +13,76 @@
 
+        /// <summary>
+        ///     The labeling convention to be used when labeling computational
+        ///     basis states.
```
```suggestion
        ///     basis states (bit string, little-endian or big-endian).
```

tcNickolas commented on src/Jupyter/KernelProperties.cs:
```diff
@@ -18,7 +22,7 @@ public static class Constants
 
             LanguageName = "qsharp",
-            LanguageVersion = "0.4",
+            LanguageVersion = "0.10",
```
Will this change update the `"version": "0.4"` field in the notebooks the next time they are edited and saved? Not objecting, just want to be prepared :-)

cgranade replied:
I believe so, yes. Ideally, we'd keep this in sync so that the metadata in notebooks is accurate, but as you can tell we haven't been great about that...

cgranade commented:
Thanks for the review, @tcNickolas! I've addressed your comment, which as a side effect dismissed your review. If you could please approve again, I'd appreciate it. Thanks!
